Python/fitnesse/html_util.py

```diff
@@ -5,38 +5,52 @@
+                    string_io=self._write_pathname_to_IO(self,string_io,suite_setup, wiki_page,"setup")
 
-            setup: WikiPage = page_crawler.get_inherited_page("SetUp", wiki_page)
+            setup: WikiPage = self._page_crawler.get_inherited_page("SetUp", wiki_page)
             if setup is not None:
-                setup_path: WikiPagePath = wiki_page.get_page_crawler().get_full_path(setup)
-                setup_path_name: str = path_parser.render(setup_path)
-                string_io.writelines(["!include -setup .", setup_path_name, "\n"])
+                string_io=self._write_pathname_to_IO(self,string_io,setup, wiki_page,"setup")
 
-        string_io.writelines([page_data.get_content()])
-        if page_data.has_attribute("Test"):
-            teardown: WikiPage = page_crawler.get_inherited_page("TearDown", wiki_page)
+        string_io.writelines([self._page_data.get_content()])
+        if self._page_data.has_attribute("Test"):
+            teardown: WikiPage = self._page_crawler.get_inherited_page("TearDown", wiki_page)
             if teardown is not None:
-                tear_down_path: WikiPagePath = wiki_page.get_page_crawler().get_full_path(teardown)
-                tear_down_path_name: str = path_parser.render(tear_down_path)
-                string_io.writelines(["!include -teardown .", tear_down_path_name, "\n"])
+                string_io=self._write_pathname_to_IO(self,string_io,teardown, wiki_page,"teardown")
             if include_suite_setup:
-                suite_teardown: WikiPage = page_crawler.get_inherited_page(SuiteResponder.SUITE_TEARDOWN_NAME, wiki_page)
+                suite_teardown: WikiPage = self._page_crawler.get_inherited_page(SuiteResponder.SUITE_TEARDOWN_NAME, wiki_page)
                 if suite_teardown is not None:
-                    page_path: WikiPagePath = wiki_page.get_page_crawler().get_full_path(suite_teardown)
-                    page_path_name: str = path_parser.render(page_path)
-                    string_io.writelines(["!include -teardown .", page_path_name, "\n"])
-
-        page_data.set_content(string_io.getvalue())
-        return page_data.get_html()
+                    string_io=self._write_pathname_to_IO(self,string_io,suite_teardown, wiki_page,"teardown")
+
+        self._page_data.set_content(string_io.getvalue())
+        return self._page_data.get_html()
+    
+    def _check_page_attribute(self, attribute: str) -> bool:
+        return self._page_data.has_attribute(attribute)
+          
+    """
+    @param type: can be "setup" or "teardown" 
+    """
+    def _write_pathname_to_IO(self,string_io: StringIO,search_page: WikiPage, wiki_page: WikiPage, type: str) -> StringIO:
+        page_path: WikiPagePath = wiki_page.get_page_crawler().get_full_path(search_page)
+        page_path_name: str = self._path_parser.render(page_path)
+        string_io.writelines(["!include -",type," .", page_path_name, "\n"])
```

Comment:
there will be whitespace between "-" and the "type". Maybe it's better to use f strings.